### PR TITLE
capybara-nplus1-fix

### PR DESCRIPTION
**Pull request by Kgomotso196**
# PROJECT TO DOs
### IN THIS PROJECT I:
1. Wrote integration tests with Capybara gem.
2. Secured app from n+1 problems.
3. Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
4. Made sure that I have used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
5. Made sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
6. Followed the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

BEFORE:
before
AFTER:
after

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    @comment7 = Comment.create(text: 'We believe in your vision!', author: @mark, post: @mark_post1)
+  end
+
+  describe 'index page' do
+    before(:example) do
+      visit user_posts_path(@mark)
+    end
+
+    it 'should render posts author information' do
+      expect(page).to have_css("img[src*='/assets/default_user.png']")
```
Comment (VICTOR-CHUKWU) [inline spec/features/post_index_spec.rb]: - [ ] all the places you have similar test of have_css is failing, kindly fix that
image

File: spec/features/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        expect(page).to have_content(comment.text)
+      end
+    end
+
+    it 'should redirect to post show page when clicking on post title' do
+      click_link @mark_post1.title.capitalize
+      expect(page).to have_current_path(user_post_path(@mark, @mark_post1))
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/features/post_show_spec.rb]: - [ ] Good job in writing those tests, but you are missing some tests like the test that asserts you can see the username of each commenter and the test that asserts you can see the comment each commentor left. kindly fix that

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Kgomotso196  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_show_spec.rb (hunk 2 above)
Comment (Kgomotso196) [reply spec/features/post_show_spec.rb]: @VICTOR-CHUKWU Thank you for a constructive feedback.

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (Kgomotso196) [reply spec/features/post_index_spec.rb]: @VICTOR-CHUKWU Thank you for a constructive feedback.

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @TEAM 

Your project is complete! Great job for taking time to make the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- capybara well-implemented
- Test well added
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**